lib/logmsg/tests/test_nvhandle_desc_array.c

```diff
@@ -28,10 +28,10 @@
 
 Test(nvhandle_desc_array, add_elements)
 {
-  NVHandleDesc desc0 = { .name = g_strdup("test0"), .flags = 0, .name_len = strlen("test0")};
-  NVHandleDesc desc1 = { .name = g_strdup("test1"), .flags = 1, .name_len = strlen("test1")};
-  NVHandleDesc desc2 = { .name = g_strdup("test2"), .flags = 2, .name_len = strlen("test2")};
-  NVHandleDesc desc3 = { .name = g_strdup("test3"), .flags = 3, .name_len = strlen("test3")};
+  NVHandleDesc desc0 = { .name = "test0", .flags = 0, .name_len = strlen("test0")};
+  NVHandleDesc desc1 = { .name = "test1", .flags = 1, .name_len = strlen("test1")};
```

Comment:
In the production code:

```
  stored.flags = 0;
  stored.name_len = len;
  stored.name = g_strdup(name);
  nvhandle_desc_array_append(self->names, &stored);
```
means we strdup the entry name, similarly to the original version of the test. Doesn't this mean that the production code still leaks? Maybe g_free is missing from `nvhandle_desc_array_free`?

Reply:
yes, you are right. i'll mark it as a wip and solve this.

Reply:
I left some explanation about the solution in the comment: https://github.com/balabit/syslog-ng/pull/2868/commits/7b5ab5077516acce86598c51eb7a97fc07808d41